### ISSUE TITLE
Xft support

### ISSUE THREAD

**Issue by aacebedo** [BurntSushi/xgb#55]
Hi !
I need to draw some text on my windows using xgb.
Drawing with X core fonts is working perfectly however as it has been deprecated for xft I was wondering how could I use it with xgb.

I didn't find any bindings for xft in go.
Any ideas ?

Regards

**Comment by BurntSushi** [comment]
Look at how Wingo does it: https://github.com/BurntSushi/wingo/tree/master/text

**Comment by aacebedo** [comment]
Thanks for the answer.
I was already digging in this direction from your example in xgbutil drawtext. However I am facing an issue with the XSurfaceSet method because I am trying to add some layer to currently existing windows and the depth of the root window is different from the depth of one of mine (terminal windows especially) This leads to a badmatch error.

In the XSurfaceSet method a pixmap is created with the depth of the root window, this pixmap is used in a changeWindowAttributesChecked call leading to the crash.

I'll continue with this approach and find a solution.

Keep you posted

**Comment by aacebedo** [comment]
So I've found two things and it is still related to the depth.

First, in the [ XDraw function](https://github.com/BurntSushi/xgbutil/blob/ad855c7130460fb6cf40af9302c649a69082a4ee/xgraphics/xsurface.go#L200 ) of an image the depth is hard coded (24), as my window has a 32 bits depth I get a BadMatch error.
When correcting this by injecting the correct issue, I still got a BadMatch error but this time in the call to the [PutImageChecked function]( https://github.com/BurntSushi/xgb/blob/5f9e7b3c49cdbee70f16f2e417cb8b473e20bc78/xproto/xproto.go#L12518)

According to [tronche.net](https://tronche.com/gui/x/xlib/graphics/XPutImage.html), the function returns this when the depth of the image is not the same as the depth of the drawable.  My drawable is a Pixmap I created with a correct depth, my image is created by the call related to the font function you pointed to me. I am currently not able to find where the depth is given.
